CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+# Kubernetes in Rust : The contributing guide.
+
+Thank you for your interest in this project. In order to contribute to this project, please get familiar with the contributing rules and the git workflow.
+
+## Git workflow :
+
+![Git workflow](https://i.imgur.com/WMNLhY8.jpg)
+
+This project is organized within 2 repositories :
+- [The main repository](https://github.com/dev-sys-do/kudo)
+- [The development repository](https://github.com/do3-2021/kudo)
+
+The development repository is a fork of the main repository. All changes made to this repository are eventually going to get through a pull request of the upstream ( main ) repository.
+
+In the development repository, you will find 3 types of branches :
+- The main branch
+- Team's branch
+- Features ( or bugfix ) branches
+
+You must code in a feature branch. This feature branch will eventually end up in a PR to your team's branch. At this level, only your team's members are required to review. This PR is annotated with **PR 1** in the image above.
+
+After accepting multiples PRs by your team's members, your team's branch will get to a point where you feel it's ready to close an issue ( more on that later ). At this point, you must open a PR from you team's branch to the main branch ( **PR 2** ). This PR must be reviewed by at least 5 reviewers before getting marked as accepted.
+
+When all the teams are satisfied with the work in the main branch, and everybody agrees on the correctness of what's been done, a new PR, from this fork to the parent repo ( the main repo ) ( **PR 3** ), will be opened to be discussed with the parent's repo maintainer. 
+
+## How to commit
+
+First, let's list the basic rules :
+- A commit message has a title ( summary ), body and footer
+- The title should be 50 characters maximum
+- Capitalise the first letter of title, don't put a period ( . ) at the end
+- Put a blank line between the title, body and footer
+- Wrap the commit body at 72 characters
+- Try to write the commit body as **bullets** ( optional )
+- Use imperative in title ( **Add** instead of **Added**, for example )
+- Describe what and why, but not how ( the purpose, not the implementation )
+- Footer contains issue number, if applicable ( #78 )
+
+> For more information about the 50/72 characters rule, please visit [this link.](https://dev.to/noelworden/improving-your-commit-message-with-the-50-72-rule-3g79#:~:text=The%2050%2F72%20Rule%20is,pulled%20out%20of%20a%20hat.)
+
+Your commits should follow the [conventional commits specification.](https://www.conventionalcommits.org/en/v1.0.0/#summary)
+
+Please be mindful of this conventionn when writing your commits. CI jobs will check if your commits follow this convention, since these commits might later be used to create changelog using parsers that understand this convention specifically.
```

Comment:
Nit: convention, not conventionn.

For the record, I personally dislike the `conventional commit`specification. I think it's confusing and messy, and it actually creates changelogs that are a bunch of commit logs.
If the goal is to be able to generate proper changlogs, then you should request that each PR adds an entry to an actual `Changelog`file. You make it the PR creator's responsibility to add something understandable to the change log.

That being said, you will be the maintainers of this project, so it's your decision.

Reply:
Nice spot for the typo !

As for the conventional commit specification, even tho it needs further discussing with the team, I think we're gonna stick with it. While your points are totally valid, I think automating changelogs will lift some weight off the team's shoulder, since our workflow is quite loaded and strict already. Although this might create some not so clear changelogs, I think it's worth giving it a try. Conventional commit is also used in a wide range of open source project, and even if it might not be the best specification, I think it's important getting used to it.

Reply:
Up to you @lucido-simon, I'm good with that decision.